pkg/k8s/certificate.go

```diff
@@ -18,81 +18,198 @@ import (
 	"context"
 	"fmt"
 	"io/ioutil"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
 	"os"
 	"path"
+	"strconv"
+	"strings"
 
 	log "github.com/sirupsen/logrus"
 	"k8s.io/apimachinery/pkg/api/errors"
 
 	"github.com/tigera/key-cert-provisioner/pkg/cfg"
 	"github.com/tigera/key-cert-provisioner/pkg/tls"
 
-	"k8s.io/api/certificates/v1beta1"
+	certV1 "k8s.io/api/certificates/v1"
+	certV1beta1 "k8s.io/api/certificates/v1beta1"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// WatchCSR Watches the CSR resource for updates and writes results to the certificate location (which should be mounted as an emptyDir)
```

Comment:
I think you can re-introduce the comment in the right place.

In general we try to have a comment for each exported func, starting with `// <FuncName> ...`